eta/core/datasets.py

```diff
@@ -1141,9 +1144,30 @@ def __init__(self, schema):
         super(Filter, self).__init__()
         self.schema = schema
 
-    def transform(self, input):
-        # @TODO implement filterin!!
-        return input
+    def _extract_video_labels(self, start_frame, end_frame, labels):
+        segment = etav.VideoLabels(schema=self.schema, filename=labels.filename)
+        for frame_id in range(start_frame, end_frame):
+            frame = labels[frame_id]
+            frame.frame_number = frame.frame_number - start_frame + 1
+            for obj in frame.objects:
+                try:
+                    segment.add_object(obj, frame.frame_number)
+                except etad.AttributeContainerSchemaError as err:
+                    logger.warn(err)
+            for attr in frame.attrs:
+                try:
+                    segment.add_frame_attribute(attr, frame.frame_number)
+                except etad.AttributeContainerSchemaError as err:
+                    logger.warn(err)
+        return segment
+
+    def transform(self, src):
+        for entry in src:
+            labels = entry.get_labels()
+            labels = self._extract_video_labels(entry.start_frame,
+                                                entry.end_frame, labels)
+            entry.set_labels(labels)
```

Comment:
Very nice!

Reply:
I do think that actual filtering should only happen once in `build`. Passing multiple  filters should just add to the schema that will be used by `build` to filter at the end.